trunk/SUAVE/Methods/Aerodynamics/Common/Fidelity_Zero/Lift/generate_propeller_wake_distribution.py

```diff
@@ -0,0 +1,231 @@
+    dt           = (2*np.pi/Na)/omega[0]
+    nts          = int(time/dt)
+    ts           = np.linspace(0,time,nts)
+    num_prop     = len(prop.origin) 
+    
+    t0           = dt*init_timestep_offset
+    start_angle  = omega[0]*t0
+    blade_angles = blade_angles  
+    
+    # define points ( control point, time step , blade number , location on blade )
+    # compute lambda and mu 
+    mean_induced_velocity  = np.mean( np.mean(va,axis = 1),axis = 1)   
+    
+    lambda_tot   =  np.atleast_2d((V_inf[:,0]  + mean_induced_velocity)).T /(omega*R)   # inflow advance ratio (page 30 Leishman)
+    mu_prop      =  np.atleast_2d(V_inf[:,2]).T /(omega*R)                              # rotor advance ratio  (page 30 Leishman) 
+    V_prop       =  np.atleast_2d(np.sqrt((V_inf[:,0]  + mean_induced_velocity)**2 + (V_inf[:,2])**2)).T
+    
+    # wake skew angle 
+    wake_skew_angle = np.arctan(mu_prop/lambda_tot)
+    
+    # reshape gamma to find the average between stations 
+    gamma_new = np.zeros((m,Na,(Nr-1)))
+    gamma_new = (gamma[:,:,:-1] + gamma[:,:,1:])*0.5
+    
+    # define empty arrays 
+    WD        = Data()
+    n         = Nr-1
+    mat1_size = (m,num_prop,nts-1,B,n)
+    WD_XA1    = np.zeros(mat1_size)  
+    WD_YA1    = np.zeros(mat1_size)  
+    WD_ZA1    = np.zeros(mat1_size)  
+    WD_XA2    = np.zeros(mat1_size)  
+    WD_YA2    = np.zeros(mat1_size)  
+    WD_ZA2    = np.zeros(mat1_size)      
+    WD_XB1    = np.zeros(mat1_size)  
+    WD_YB1    = np.zeros(mat1_size)  
+    WD_ZB1    = np.zeros(mat1_size)  
+    WD_XB2    = np.zeros(mat1_size)   
+    WD_YB2    = np.zeros(mat1_size)   
+    WD_ZB2    = np.zeros(mat1_size)     
+    WD_GAMMA  = np.zeros(mat1_size)     
+    
+    mat2_size = (m,num_prop*(nts-1)*B*n)
+    WD.XA1    = np.zeros(mat2_size)
+    WD.YA1    = np.zeros(mat2_size)
+    WD.ZA1    = np.zeros(mat2_size)
+    WD.XA2    = np.zeros(mat2_size)
+    WD.YA2    = np.zeros(mat2_size)
+    WD.ZA2    = np.zeros(mat2_size)   
+    WD.XB1    = np.zeros(mat2_size)
+    WD.YB1    = np.zeros(mat2_size)
+    WD.ZB1    = np.zeros(mat2_size)
+    WD.XB2    = np.zeros(mat2_size)
+    WD.YB2    = np.zeros(mat2_size)
+    WD.ZB2    = np.zeros(mat2_size) 
+    
+    VD.Wake       = Data()
+    mat3_size     = (num_prop,(nts-1),B,n)
+    VD.Wake.XA1   = np.zeros(mat3_size) 
+    VD.Wake.YA1   = np.zeros(mat3_size) 
+    VD.Wake.ZA1   = np.zeros(mat3_size) 
+    VD.Wake.XA2   = np.zeros(mat3_size) 
+    VD.Wake.YA2   = np.zeros(mat3_size) 
+    VD.Wake.ZA2   = np.zeros(mat3_size)    
+    VD.Wake.XB1   = np.zeros(mat3_size) 
+    VD.Wake.YB1   = np.zeros(mat3_size) 
+    VD.Wake.ZB1   = np.zeros(mat3_size) 
+    VD.Wake.XB2   = np.zeros(mat3_size) 
+    VD.Wake.YB2   = np.zeros(mat3_size) 
+    VD.Wake.ZB2   = np.zeros(mat3_size)     
+     
+    for i in range(num_prop): 
+        Gamma  = np.zeros((m,nts-1,B,n))   
+        num = int(Na/B)  
+        for t_idx in range(nts-1): 
```

Comment:
This double for loop can be vectorized. Effectively there is a triple loop here.

Reply:
I vectorized the double for loop. The only one remaining is needed as the propeller locations are different and it propagates  to sub-functions

Reply:
Awesome! Yeah, haha, that propeller location loop isn't going anywhere.

Once it's pushed and I see the green check I'll approve. Maybe you can take a much needed break from long PR's for a bit? haha